cdf_fabric_replicator/data_modeling.py

```diff
@@ -63,14 +67,23 @@ def run(self) -> None:
                 self.config.extractor.poll_time - elapsed_time, 0
             )  # 900s = 15min
             if sleep_time > 0:
-                logging.debug(f"Sleep for {sleep_time} seconds")
+                self.logger.debug(f"Sleep for {sleep_time} seconds")
                 time.sleep(sleep_time)
```

Comment:
stop_event.wait() might be even better since we have a cancellation token. 
```suggestion
                self.stop_event.wait(sleep_time)
```

Reply:
Good idea, done!